src/main/java/frc/robot/vision/filters/VisionFilters.java

```diff
@@ -0,0 +1,43 @@
+package frc.robot.vision.filters;
+
+import edu.wpi.first.math.geometry.Rotation2d;
+import frc.robot.constants.Field;
+import frc.robot.vision.rawdata.AprilTagVisionData;
+import frc.robot.vision.rawdata.VisionData;
+
+public class VisionFilters {
```

Comment:
Create a generic ToleranceFilter which implements a value being in range of the tolerance, and implement the rest of the filters with it - i.e.

```java
Filter<VisionData> isPitchInTolerance(Rotation2d pitchTolerance) {
    return new ToleranceFilter<Rotation2d>(pitchTolerance.toRadians(), visionData.getEstimatedPose().getRotation().getY());
}
```

Reply:
i think its too much for just 6 functions, and it will not reduce lines as seen in your example. i dont see why this is beneficial